crates/plugins/mediator-coordination/src/model/stateful/entity.rs

```diff
@@ -47,3 +50,33 @@ pub struct VerificationMaterial {
     pub format: i32,
     pub value: String,
 }
+
+impl Identifiable for Connection {
+    fn id(&self) -> Option<ObjectId> {
+        self.id
+    }
+
+    fn set_id(&mut self, id: ObjectId) {
+        self.id = Some(id);
+    }
+}
+
+impl Identifiable for RoutedMessage {
+    fn id(&self) -> Option<ObjectId> {
+        self.id
+    }
+
+    fn set_id(&mut self, id: ObjectId) {
+        self.id = Some(id);
+    }
+}
+
+impl Identifiable for Secrets {
+    fn id(&self) -> Option<ObjectId> {
+        self.id.clone() // Adjust if making it optional
```

Comment:
Why cloning? Also please remove all the unnecessary comments.

Reply:
To create a new instance of the Option<ObjectId>
But i am to remove